radsat-sk/operation/subsystems/RBattery.c

```diff
@@ -0,0 +1,318 @@
+static uint32_t batteryTemperatureCommandBytes[4] = {
+		0x10E308,		// Motherboard Temperature
+		0x10E398,		// Daughterboard 1 Temperature
+		0x10E3A8,		// Daughterboard 2 Temperature
+		0x10E3B8		// Daughterboard 3 Temperature
+};
+
+/***************************************************************************************************
+                                       PRIVATE FUNCTION STUBS
+***************************************************************************************************/
+
+static int batteryTalk(uint8_t* command, uint8_t* response);
+static int checkSafeFlag(uint8_t* safeFlag);
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Request and store all of the relevant data (voltage, current, temperature) from the battery board
+ * 		and battery daughter boards
+ * @return A battery_status_t object containing all of the telemetry of the battery board
+ */
+int batteryTelemetry(battery_status_t* dataStorage) {
+
+	// Create temporary variables for sending and receiving I2C data
+	uint8_t command[BATTERY_TELEM_COMMAND_LENGTH] = {0};
+	uint8_t response[BATTERY_RESPONSE_LENGTH] = {0};
+
+	// Create a temporary array for calculated voltage and current before transferring into the battery_status_t structure
+	float storedData[NUMBER_OF_CURRENT_COMMANDS] = {0};
+
+	// Create a temporary array for calculated temperatures before transferring into the battery_status_t structure
+	float storedTemperatureData[NUMBER_OF_TEMP_COMMANDS] = {0};
+
+
+	// Send 3 commands to get ADC output voltage readings from the Battery
+	for (int i = 0; i < NUMBER_OF_VOLTAGE_COMMANDS; i = i + 1) {
+		memset(command, 0, sizeof(command));
+		memcpy(command, &batteryVoltageCommandBytes[i], BATTERY_TELEM_COMMAND_LENGTH);
+
+		memset(response, 0, sizeof(response));
+
+		int error = batteryTalk(command, response);
+
+		if (error != SUCCESS){
+			return error;
+		}
+		memset(&storedData[i], 0, sizeof(storedData[i]));
+		memcpy(&storedData[i], response, BATTERY_RESPONSE_LENGTH);
+	}
+
+	// Get ADC Output Voltages
+	dataStorage->outputVoltageBatteryBus = ADC_COUNT_TO_BATTERY_BUS_OUTPUT_VOLTAGE * storedData[0];
+	dataStorage->outputVoltage5VBus = ADC_COUNT_TO_5V_BUS_OUTPUT_VOLTAGE * storedData[1];
+	dataStorage->outputVoltage3V3Bus = ADC_COUNT_TO_3V3_BUS_OUTPUT_VOLTAGE * storedData[2];
+
+
+	// Send 4 commands to get ADC output current readings from the Battery
+	for (int i = 0; i < NUMBER_OF_CURRENT_COMMANDS; i = i + 1) {
+		memset(command, 0, sizeof(command));
+		memcpy(command, &batteryCurrentCommandBytes[i], BATTERY_TELEM_COMMAND_LENGTH);
+
+		memset(response, 0, sizeof(response));
+
+		int error = batteryTalk(command, response);
+
+		if (error != SUCCESS) {
+			return error;
+		}
+
+		memset(&storedData[i], 0, sizeof(&storedData[i]));
```

Comment:
Might have to change the second argument to just `sizeof(storedData)` instead of `sizeof(&storedData[i])`. I think they're technically the same but the former is more clearer in my opinion